src/lib.rs

```diff
@@ -274,21 +274,18 @@ fn config_files_and_rev_for_repo(
     // - otherwise try naive http requests first,
     // - and then finally clone the repo and look
     let local_git_dir = local_repo_dir.join(".git");
-    if local_git_dir.exists() {
-        let rev = get_git_rev(&local_repo_dir).map_err(ConfigFetchIssue::GitFail)?;
-        let configs = get_config_paths(&local_repo_dir).ok_or(ConfigFetchIssue::NoConfigFound)?;
-        return Ok((configs, rev));
-    }
+    let skip_naive = local_git_dir.exists();
 
-    let naive = config_file_and_rev_from_remote_naive(repo_url).map(|(p, rev)| (vec![p], rev));
-    // if not found, try checking out and looking; otherwise return the result
-    if !matches!(naive, Err(ConfigFetchIssue::NoConfigFound)) {
-        naive
-    } else {
-        let configs = config_files_from_local_checkout(repo_url, &local_repo_dir)?;
-        let rev = get_git_rev(&local_repo_dir).map_err(ConfigFetchIssue::GitFail)?;
-        Ok((configs, rev))
+    if !skip_naive {
+        let naive = config_file_and_rev_from_remote_naive(repo_url).map(|(p, rev)| (vec![p], rev));
```

Comment:
nit: I don't find the naive naming informative, maybe we could be more descriptive? - for example, `try_standard_config_location` or some such